src/internal/components/chart-wrapper/index.tsx

```diff
@@ -35,25 +36,59 @@ export const ChartWrapper = forwardRef(
       onBlur,
       contentClassName,
       contentMinHeight,
+      fitHeight,
       ...props
     }: ChartWrapperProps,
     ref: React.Ref<HTMLDivElement>
   ) => {
     const baseProps = getBaseProps(props);
+
+    const filtersNode = (defaultFilter || additionalFilters) && (
+      <InternalBox className={styles['filter-container']} margin={{ bottom: 'l' }}>
+        <InternalSpaceBetween
+          size="l"
+          direction="horizontal"
+          className={clsx({ [styles['has-default-filter']]: !!defaultFilter })}
+        >
+          {defaultFilter}
+          {additionalFilters}
+        </InternalSpaceBetween>
+      </InternalBox>
+    );
+
+    const legendNode = legend && <InternalBox margin={{ top: 'm' }}>{legend}</InternalBox>;
+
+    if (fitHeight) {
```

Comment:
I see many `if(fitHeight) {...}` code duplication patterns now. Does it have to do anything with performance optimization?